ui/components/app/confirm/info/row/row.tsx

```diff
@@ -81,69 +82,96 @@ export const ConfirmInfoRow: React.FC<ConfirmInfoRowProps> = ({
-        ...style,
-      }}
-    >
-      {copyEnabled && <CopyIcon copyText={copyText ?? ''} />}
+  collapsible,
+}) => {
+  const [expanded, setExpanded] = useState(!collapsible);
+
+  return (
+    <ConfirmInfoRowContext.Provider value={{ variant }}>
       <Box
+        data-testid={dataTestId}
+        className="confirm-info-row"
         display={Display.Flex}
-        flexDirection={FlexDirection.Row}
-        justifyContent={JustifyContent.center}
-        alignItems={AlignItems.flexStart}
-        color={color}
+        flexDirection={collapsible ? FlexDirection.Column : FlexDirection.Row}
+        justifyContent={JustifyContent.spaceBetween}
+        flexWrap={FlexWrap.Wrap}
+        alignItems={collapsible ? AlignItems.flexStart : AlignItems.center}
+        backgroundColor={BACKGROUND_COLORS[variant]}
+        borderRadius={BorderRadius.LG}
+        marginTop={2}
+        marginBottom={2}
+        paddingLeft={2}
+        paddingRight={copyEnabled ? 5 : 2}
+        color={TEXT_COLORS[variant] as TextColor}
+        style={{
+          overflowWrap: OverflowWrap.Anywhere,
+          minHeight: '24px',
+          position: 'relative',
+          ...style,
+        }}
       >
-        <Box display={Display.Flex} alignItems={AlignItems.center}>
-          <Text variant={TextVariant.bodyMdMedium} color={TextColor.inherit}>
-            {label}
-          </Text>
-          {labelChildren}
-          {!labelChildren && tooltip?.length && (
-            <Tooltip
-              position="bottom"
-              title={tooltip}
-              style={{ display: 'flex' }}
-            >
-              <Icon
-                name={TOOLTIP_ICONS[variant]}
-                marginLeft={1}
-                color={TOOLTIP_ICON_COLORS[variant] as unknown as IconColor}
-                size={IconSize.Sm}
-                {...(dataTestId
-                  ? { 'data-testid': `${dataTestId}-tooltip` }
-                  : {})}
-              />
-            </Tooltip>
-          )}
+        {copyEnabled && (
+          <CopyIcon
+            copyText={copyText ?? ''}
+            style={{ right: collapsible ? 32 : 0, top: 4 }}
+            color={IconColor.iconMuted}
+          />
+        )}
+        {collapsible && (
+          <Icon
+            color={IconColor.iconMuted}
+            name={expanded ? IconName.Collapse : IconName.Expand}
+            size={IconSize.Sm}
+            style={{
+              cursor: 'pointer',
+              position: 'absolute',
+              right: 8,
+              top: 4,
+            }}
+            onClick={() => setExpanded(!expanded)}
+            data-testid="sectionCollapsibleButton"
+          />
+        )}
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Row}
+          justifyContent={JustifyContent.center}
+          alignItems={AlignItems.flexStart}
+          color={color}
+        >
+          <Box display={Display.Flex} alignItems={AlignItems.center}>
```

Comment:
While it's nice to add configuration to the core row, this adds a lot of logic and complexity, so would it be better to create an alternate wrapper component that potentially inherits the same properties of row? Or a wrapper component?

Or at the very least, should we break this very large schema up with local components?

Reply:
Changes are required in info row requires to support collapsible row, it is hard to only wrap into row and make it collapsible.

Reply:
In that case, could we split this for readability into some local components (in this file) such as `Container`, `CopyIcon`, `Tooltip`, `Content`, `CollapsibleIcon` etc?